### PR TITLE
[RFR] Fixing comments and filters

### PR DESCRIPTION
**Pull request by jpetitcolas**
- [x] Fix comments addition and edition forms
- [x] Fix filters
- [x] Add some tests
- [x] Fix last behavior (see comments)

### CODE REVIEW

File: src/mui/list/List.js [hunk 1]
```diff
@@ -37,6 +37,13 @@ class List extends Component {
         }
     }
 
+    componentWillUpdate(nextProps) {
+        const changedProps = this.getChangedProps(nextProps);
+        Object.keys(changedProps).forEach(changedPropName => {
+            this.setFilter(changedPropName, changedProps[changedPropName]);
```
Comment (fzaninotto) [inline src/mui/list/List.js]: You're doing a `setFilter` for *all* props... some props don't relate to filters

File: src/mui/list/List.js [hunk 2]
```diff
@@ -60,11 +67,25 @@ class List extends Component {
...
     }
 
+    /**
+     * check which props has changed, and return an object whose keys is
+     * changed prop name and value is the new prop value.
+     */
+    getChangedProps = ({ filters }) => Object.keys(filters)
+        .reduce((changedProps, filterName) => {
+            const updatedProps = { ...changedProps };
+            if (filters[filterName] !== this.props.filters[filterName]) {
```
Comment (fzaninotto) [inline src/mui/list/List.js]: You don't need to diff yourself - React will do it.

File: src/mui/list/List.js (hunk 1 above)
Comment (fzaninotto) [inline src/mui/list/List.js]: don't forget to add `debounce` here

**Comment by jpetitcolas** [comment]
PR almost done. Fixing a last issue related to filters:

* Go on Posts list
* Refresh the page (F5)
* Search for "hello": no results.
* Erase the "hello" search: still no results.

**Comment by jpetitcolas** [comment]
Switching to RFR.

File: src/mui/detail/Create.js [hunk 3]
```diff
@@ -19,9 +19,8 @@ class Create extends Component {
         return location.pathname.split('/').slice(0, -1).join('/');
     }
 
-    handleSubmit(event) {
-        event.preventDefault();
-        this.props.crudCreate(this.props.resource, this.state.record, this.getBasePath());
+    handleSubmit(record) {
```
Comment (djhi) [inline src/mui/detail/Create.js]:
```js
handleSubmit = event => {
...
}
```

and you won't need the bind in constructor anymore

File: src/mui/list/FilterForm.js [hunk 4]
```diff
@@ -1,50 +1,31 @@
...
+import { Field, reduxForm } from 'redux-form';
 import { CardText } from 'material-ui/Card';
 import IconButton from 'material-ui/IconButton';
 import ActionHide from 'material-ui/svg-icons/action/highlight-off';
 
 export class FilterForm extends Component {
     constructor(props) {
         super(props);
-        this.handleChange = this.handleChange.bind(this);
         this.handleHide = this.handleHide.bind(this);
```
Comment (djhi) [inline src/mui/list/FilterForm.js]: same here, no need for bind

File: src/mui/list/List.js [hunk 5]
```diff
@@ -5,20 +5,21 @@ import { Card, CardTitle, CardActions } from 'material-ui/Card';
...
-class List extends Component {
+export class List extends Component {
     constructor(props) {
         super(props);
         this.state = {};
         this.updateSort = this.updateSort.bind(this);
         this.setPage = this.setPage.bind(this);
-        this.setFilter = this.setFilter.bind(this);
         this.showFilter = this.showFilter.bind(this);
         this.hideFilter = this.hideFilter.bind(this);
```
Comment (djhi) [inline src/mui/list/List.js]: same for all of them

**Comment by jpetitcolas** [comment]
Code review applied.

**Pull request merged**